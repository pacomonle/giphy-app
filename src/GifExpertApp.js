import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({defaultCategories=[]}) => {

  // const initialState = ['Rick and Morty']
  // const [categories, setCategories] = useState(initialState)

const [categories, setCategories] = useState(defaultCategories)
   /*
   const handleAdd = () => {
     // setCategories([...categories, 'HunterXHunter'])
     setCategories(cat => [...cat, 'HunterXHunter'])
   }

*/
    return (
        <>
           <h1>GifExpertApp</h1>
           <AddCategory setCategories={setCategories}></AddCategory>
           <hr />
           
           <ol>
               {
                  categories.map((category, i) =>( 
                    <GifGrid key={i} category={category}></GifGrid> 
                  ))
               }
           </ol>
        </>
    )
}

export default GifExpertApp

