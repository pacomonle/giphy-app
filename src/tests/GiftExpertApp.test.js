import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GiftExperApp></GiftExperApp>', () => {
    

    test('should mostrar componente <GiftExperApp /> ', () => {
        const wrapper = shallow(<GifExpertApp></GifExpertApp>)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should mostrar una lsita de categorias', () => {
        const categories = ['Dragon Ball', 'One Punch']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    })
    
    
})
