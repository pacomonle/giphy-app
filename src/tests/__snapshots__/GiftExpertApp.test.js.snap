// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExperApp></GiftExperApp> should mostrar componente <GiftExperApp />  1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GiftExperApp></GiftExperApp> should mostrar una lsita de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="0"
    />
    <GifGrid
      category="One Punch"
      key="1"
    />
  </ol>
</Fragment>
`;
